chore: Merge branch `master` into `rc/v11.0.0`

cpcallen opened the pull request:
## The basics

- [X] I [validated my changes](https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

## The details
### Proposed Changes

Merge branch `master` into `rc/v11.0.0`.

### Reason for Changes

To pull in certain changes (e.g. #2339, #2344) that are needed in both branches.

### Additional Information

Please verify manual resolution of conflicting changes to `plugins/dev-tools/src/screenshot.js` is correct.

**CLOSE THIS PR BY MERGE COMMIT**

dependabot[bot] committed:
chore(deps-dev): Bump webpack-dev-middleware (#2258)

Bumps [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) from 5.3.3 to 5.3.4.
- [Release notes](https://github.com/webpack/webpack-dev-middleware/releases)
- [Changelog](https://github.com/webpack/webpack-dev-middleware/blob/v5.3.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-middleware/compare/v5.3.3...v5.3.4)

---
updated-dependencies:
- dependency-name: webpack-dev-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump webpack-dev-middleware from 5.3.1 to 5.3.4 in /plugins/dev-scripts (#2262)

* chore(deps): Bump webpack-dev-middleware in /plugins/dev-scripts

Bumps [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) from 5.3.1 to 5.3.4.
- [Release notes](https://github.com/webpack/webpack-dev-middleware/releases)
- [Changelog](https://github.com/webpack/webpack-dev-middleware/blob/v5.3.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-middleware/compare/v5.3.1...v5.3.4)

---
updated-dependencies:
- dependency-name: webpack-dev-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore: update package-locks [dependabot skip]

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: blockly[bot] <blockly-github-robot@google.com>

dependabot[bot] committed:
chore(deps-dev): Bump webpack-dev-middleware (#2260)

Bumps [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) from 5.3.3 to 5.3.4.
- [Release notes](https://github.com/webpack/webpack-dev-middleware/releases)
- [Changelog](https://github.com/webpack/webpack-dev-middleware/blob/v5.3.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-middleware/compare/v5.3.3...v5.3.4)

---
updated-dependencies:
- dependency-name: webpack-dev-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump webpack-dev-middleware in /examples/sample-app (#2264)

Bumps [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) from 5.3.3 to 5.3.4.
- [Release notes](https://github.com/webpack/webpack-dev-middleware/releases)
- [Changelog](https://github.com/webpack/webpack-dev-middleware/blob/v5.3.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-middleware/compare/v5.3.3...v5.3.4)

---
updated-dependencies:
- dependency-name: webpack-dev-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mobyw committed:
fix(doc): correct image filename to make it can be displayed in the doc (#2269)

rachel-fenichel committed:
chore: stop testing on node 16 (#2274)

BeksOmega committed:
fix: readme for the dynamic connection previewer (#2272)

dependabot[bot] committed:
chore(deps-dev): Bump ip from 1.1.5 to 1.1.9 in /examples/blockly-rtc (#2213)

Bumps [ip](https://github.com/indutny/node-ip) from 1.1.5 to 1.1.9.
- [Commits](https://github.com/indutny/node-ip/compare/v1.1.5...v1.1.9)

---
updated-dependencies:
- dependency-name: ip
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump follow-redirects (#2248)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.15.4 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.15.4...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump follow-redirects (#2253)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.15.4 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.15.4...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump webpack-dev-middleware in /examples/sample-app-ts (#2259)

Bumps [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) from 5.3.3 to 5.3.4.
- [Release notes](https://github.com/webpack/webpack-dev-middleware/releases)
- [Changelog](https://github.com/webpack/webpack-dev-middleware/blob/v5.3.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-middleware/compare/v5.3.3...v5.3.4)

---
updated-dependencies:
- dependency-name: webpack-dev-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump webpack-dev-middleware (#2266)

Bumps [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) from 5.3.3 to 5.3.4.
- [Release notes](https://github.com/webpack/webpack-dev-middleware/releases)
- [Changelog](https://github.com/webpack/webpack-dev-middleware/blob/v5.3.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-middleware/compare/v5.3.3...v5.3.4)

---
updated-dependencies:
- dependency-name: webpack-dev-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump express in /examples/sample-app-ts (#2277)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump express in /examples/blockly-rtc (#2276)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump express in /examples/sample-app (#2280)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump express in /examples/custom-generator-codelab (#2284)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump express from 4.18.2 to 4.19.2 in /plugins/dev-scripts (#2282)

* chore(deps): Bump express from 4.18.2 to 4.19.2 in /plugins/dev-scripts

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* chore: update package-locks [dependabot skip]

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: blockly[bot] <blockly-github-robot@google.com>

dependabot[bot] committed:
chore(deps-dev): Bump follow-redirects (#2246)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.15.4 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.15.4...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump follow-redirects in /examples (#2251)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.15.4 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.15.4...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump follow-redirects in /examples/sample-app-ts (#2252)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.15.4 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.15.4...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump webpack-dev-middleware (#2261)

Bumps [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) from 5.3.3 to 5.3.4.
- [Release notes](https://github.com/webpack/webpack-dev-middleware/releases)
- [Changelog](https://github.com/webpack/webpack-dev-middleware/blob/v5.3.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-middleware/compare/v5.3.3...v5.3.4)

---
updated-dependencies:
- dependency-name: webpack-dev-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump webpack-dev-middleware in /examples/blockly-react (#2265)

Bumps [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) from 5.3.3 to 5.3.4.
- [Release notes](https://github.com/webpack/webpack-dev-middleware/releases)
- [Changelog](https://github.com/webpack/webpack-dev-middleware/blob/v5.3.4/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-middleware/compare/v5.3.3...v5.3.4)

---
updated-dependencies:
- dependency-name: webpack-dev-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump express in /examples/blockly-angular (#2279)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump express in /examples/keyboard-navigation-codelab (#2281)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): Bump express in /examples/custom-renderer-codelab (#2275)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump express in /examples/blockly-react (#2283)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

github-actions[bot] committed:
Publish

 - @blockly/block-dynamic-connection@0.6.1
 - @blockly/block-plus-minus@7.0.3
 - @blockly/block-shareable-procedures@4.1.2
 - @blockly/workspace-content-highlight@4.0.13
 - @blockly/continuous-toolbox@5.0.14
 - @blockly/plugin-cross-tab-copy-paste@5.0.7
 - @blockly/dev-tools@7.1.7
 - @blockly/disable-top-blocks@0.4.11
 - @blockly/field-angle@3.0.13
 - @blockly/field-bitmap@4.0.14
 - @blockly/field-colour@3.0.13
 - @blockly/field-colour-hsv-sliders@4.2.2
 - @blockly/field-date@8.0.13
 - @blockly/field-dependent-dropdown@3.0.13
 - @blockly/field-grid-dropdown@4.0.12
 - @blockly/field-multilineinput@3.0.14
 - @blockly/field-slider@6.1.6
 - @blockly/fixed-edges@4.0.12
 - @blockly/keyboard-navigation@0.5.9
 - @blockly/plugin-modal@6.0.11
 - @blockly/plugin-scroll-options@5.0.13
 - @blockly/shadow-block-converter@5.0.2
 - @blockly/plugin-strict-connection-checker@4.0.12
 - @blockly/suggested-blocks@4.1.7
 - @blockly/theme-deuteranopia@5.0.6
 - @blockly/toolbox-search@1.2.7
 - @blockly/plugin-typed-variable-modal@7.0.11
 - @blockly/workspace-backpack@5.3.5
 - @blockly/workspace-minimap@0.1.15
 - @blockly/plugin-workspace-search@8.1.4
 - @blockly/zoom-to-fit@5.0.14

NeilFraser committed:
chore: Remove colour and multiline from devsite demo (#2294)

Also update JS-Interpreter to latest version.

rachel-fenichel committed:
chore: allow const variables to have UPPER_CASE names (#2300)

rachel-fenichel committed:
chore: remove references to plugin-ified fields and blocks (#2302)

* fix: remove references to colour fields and blocks (#2241)

* chore: delete uses of colour blocks from most other plugins

* chore: delete colour blocks from shadow block converter

* chore: stop using colour fields in block-test plugin

* fix: param order

* chore: remove uses of the multiline block or field in other plugins (#2239)

rachel-fenichel committed:
feat!: add block definitions to fields (#2299)

* feat!: add block definitions to colour field plugin (#2162)

* feat: export functions to register some angle, colour, and multiline input fields

* feat: add block definitions to colour and multiline input fields

* feat: add block generators for the colour_picker block

* fix: use Blockly.common.defineBlocksWithJsonArray

* feat: add block generators for colour_random block

* feat: add generators type and standardize exports for block files

* chore: update to blockly@beta and fix types

* chore: move all colour blocks to separate files and add more generator-related types

* feat: finish adding block code generators for colour blocks

* fix: PR feedback

* fix: remove immediate registration of blocks and fields in field_colour

* chore: use named imports and numbered TODOs

* feat: add usage information about blocks to README

* chore: revert changes outside of field_colour

* chore: clean up tsdoc and exports

* cgire: clean up README

* chore: respond to PR feedback on names and comments

* feat(tests): add and improve tests

* chore(format): run formatter

* fix: respond to PR feedback about names and file structure

* fix: allow const variables to have UPPER_CASE names

* fix: respond to PR feedback

* chore: format

* fix: line length

* feat!: add block definition to multiline text field plugin (#2202)

* chore: force-install blockly 10.4.0-beta.1 for development

* feat: add text_multiline block and associated generators

* feat: update test page to use new block and field

* feat: README

* chore: respond to PR feedback

BREAKING CHANGE: The multiline text input field no longer registers itself on load. The developer must either manually register the field or install blocks, which will install the field. This is part of a move to have no side effects in field and block definitions, so that tree-shaking can remove unwanted fields and blocks.

* chore: format (#2221)

* feat!: Add registration function to field_angle and make it no longer install on file load (#2211)

* feat!: Add registration function to field_angle and make it no longer install on file load

* chore: formatting

BREAKING CHANGE: The angle field no longer registers itself on load. The developer must manually register the field. This is part of a move to have no side effects in field and block definitions, so that tree-shaking can remove unwanted fields and blocks.

* feat: add tests for per-block generators in field-colour (#2220)

* feat: add unit tests for generators

* fix: imports in tests

* chore: format

* feat: ignore golden files in plugin tests

* chore: fix lint

* feat: ignore golden files for linting

* fix: revert addition of fs and path packages

* fix: remove suite.only to make all tests run

* chore: handle review feedback

* fix: only import what you need in colour blocks

* feat: add generator tests for the text_multiline block (#2232)

* feat: add generator tests for the text_multiline block

* fix: updated test string to include multiple types of quotes

* fix: code style in generators for field colour blocks (#2233)

* feat!: update blockly version to 10.4.3 for colour and multilineinput (#2296)

* feat!: update blockly version to 10.4.3 for colour and multilineinput

* chore!: update peer dependencies

* chore: update package-lock.jsons

* chore: fix dependencies

github-actions[bot] committed:
Publish

 - @blockly/block-dynamic-connection@0.6.2
 - @blockly/block-plus-minus@7.0.4
 - @blockly/block-shareable-procedures@4.1.3
 - @blockly/block-test@5.1.1
 - @blockly/workspace-content-highlight@4.0.14
 - @blockly/continuous-toolbox@5.0.15
 - @blockly/plugin-cross-tab-copy-paste@5.0.8
 - @blockly/dev-tools@7.1.8
 - @blockly/disable-top-blocks@0.4.12
 - @blockly/field-angle@4.0.0
 - @blockly/field-bitmap@4.0.15
 - @blockly/field-colour@4.0.0
 - @blockly/field-colour-hsv-sliders@4.2.3
 - @blockly/field-date@8.0.14
 - @blockly/field-dependent-dropdown@3.0.14
 - @blockly/field-grid-dropdown@4.0.13
 - @blockly/field-multilineinput@4.0.0
 - @blockly/field-slider@6.1.7
 - @blockly/fixed-edges@4.0.13
 - @blockly/keyboard-navigation@0.5.10
 - @blockly/plugin-modal@6.0.12
 - @blockly/plugin-scroll-options@5.0.14
 - @blockly/shadow-block-converter@5.0.3
 - @blockly/plugin-strict-connection-checker@4.0.13
 - @blockly/suggested-blocks@4.1.8
 - @blockly/theme-dark@6.0.6
 - @blockly/theme-deuteranopia@5.0.7
 - @blockly/theme-highcontrast@5.0.6
 - @blockly/theme-modern@5.0.6
 - @blockly/theme-tritanopia@5.0.6
 - @blockly/toolbox-search@1.2.8
 - @blockly/plugin-typed-variable-modal@7.0.12
 - @blockly/workspace-backpack@5.3.6
 - @blockly/workspace-minimap@0.1.16
 - @blockly/plugin-workspace-search@8.1.5
 - @blockly/zoom-to-fit@5.0.15

maribethb committed:
fix: fix invalid url in scroll options (#2310)

mobyw committed:
fix: remove code of removed colour toolbox (#2308)

* fix: devsite init script

* fix: devsite style

cpcallen committed:
fix(migration): Ignore non-renamed modules (#2309)

* fix(migration): Ignore non-renamed modules
  + add a test for this.

* chore(migration): Remove log statement

mobyw committed:
chore: optimize devsite's toolbox css config and localized names injection (#2312)

* feat: use css config in devsite toolbox

* feat: rename `functions` with `procedures` to correspond to msgs keys

* feat: optimize localized string filling approach in devsite

* feat: remove colour block style of devsite

* chore(format): format with prettier

* chore(format): format another file to pass check

rachel-fenichel committed:
chore: add readme media for plugin fields (#2318)

* chore: add readme media for colour field

* chore: add readme-media for angle field

* chore: add readme-media for multiline input field

* chore: delete unused image

rachel-fenichel committed:
chore: update colour readme to include more information about field customization (#2319)

* chore: update field-colour readme to include information from devsite

* chore: line length

* chore: show colourOptions in colour field sample code

* chore: format

* chore: respond to review feedback

github-actions[bot] committed:
Publish

 - @blockly/field-angle@4.0.1
 - @blockly/field-colour@4.0.1
 - @blockly/field-colour-hsv-sliders@4.2.4
 - @blockly/field-multilineinput@4.0.1
 - @blockly/migrate@2.0.5
 - @blockly/plugin-scroll-options@5.0.15

gonfunko committed:
chore: Clean up and convert field-bitmap to Typescript (#2227)

* chore: Add a tsconfig

* chore: Rename JS files to TS

* chore: Convert to TS

* refactor: Clean up comments and minorly refactor functionality.

* chore: Run format.

* chore: Clean up comments and remove trailing underscore.

* fix: disable eslint for overridden methods

* fix: disable eslint in such a way that prettier doesn't whine about it

github-actions[bot] committed:
Publish

 - @blockly/field-bitmap@4.0.16

Shivam7-1 committed:
fix: update screenshot.js DOM text reinterpreted as HTML

Shivam7-1 committed:
fix: Update index.js DOM text reinterpreted as HTML (#2329)

Shivam7-1 committed:
fix: Update screenshot.js DOM text reinterpreted as HTML  (#2328)

Shivam7-1 committed:
chore: Change innerHTML to textContent

No security issue.

ashmichheda committed:
fix: add typed variable plugin (#2335)

github-actions[bot] committed:
Publish

 - @blockly/block-dynamic-connection@0.6.3
 - @blockly/block-plus-minus@7.0.5
 - @blockly/block-shareable-procedures@4.1.4
 - @blockly/workspace-content-highlight@4.0.15
 - @blockly/continuous-toolbox@5.0.16
 - @blockly/plugin-cross-tab-copy-paste@5.0.9
 - @blockly/dev-tools@7.1.9
 - @blockly/disable-top-blocks@0.4.13
 - @blockly/field-angle@4.0.2
 - @blockly/field-bitmap@4.0.17
 - @blockly/field-colour@4.0.2
 - @blockly/field-colour-hsv-sliders@4.2.5
 - @blockly/field-date@8.0.15
 - @blockly/field-dependent-dropdown@3.0.15
 - @blockly/field-grid-dropdown@4.0.14
 - @blockly/field-multilineinput@4.0.2
 - @blockly/field-slider@6.1.8
 - @blockly/fixed-edges@4.0.14
 - @blockly/keyboard-navigation@0.5.11
 - @blockly/plugin-modal@6.0.13
 - @blockly/plugin-scroll-options@5.0.16
 - @blockly/shadow-block-converter@5.0.4
 - @blockly/plugin-strict-connection-checker@4.0.14
 - @blockly/suggested-blocks@4.1.9
 - @blockly/toolbox-search@1.2.9
 - @blockly/plugin-typed-variable-modal@7.0.13
 - @blockly/workspace-backpack@5.3.7
 - @blockly/workspace-minimap@0.1.17
 - @blockly/plugin-workspace-search@8.1.6
 - @blockly/zoom-to-fit@5.0.16

gonfunko committed:
fix: only select the first category when the flyout is shown when there is no selection (#2337)

github-actions[bot] committed:
Publish

 - @blockly/continuous-toolbox@5.0.17

sancodes committed:
fix: add boolean to dispose function call  (#2341)

* fix: add boolean to dispose function call (#2217)

* fix: replace block connections

maribethb committed:
chore: fix multiline package lock (#2340)

cpcallen committed:
fix: Fix incorrect Blockly imports (#2339)

We have for some time promulgated

   import * as Blockly from 'blockly/core';

as the correct way to import Blockly in our documentation.
Update our own plugins and templates to match.

Because we have been packaging Blockly until now only as CJS
modules, and CJS modules always provide an (unnamed) default
export consisting of the module.exports object, importing
Blockly with

    import Blockly from 'blockly/core';

(note no "*") has worked fine (and similarly for all the
other entrypoints).

As the sources for the entrypoints (core/blockly.ts,
blocks/blocks.ts, etc.) do not explicitly declare a default
export, however, this should not be relied upon: when we start
shipping Blockly as pure-ESM there will be no default export
and so "import Blockly from …" will stop working.

github-actions[bot] committed:
Publish

 - @blockly/block-dynamic-connection@0.6.4
 - @blockly/block-plus-minus@7.0.6
 - @blockly/block-shareable-procedures@4.1.5
 - @blockly/block-test@5.1.2
 - @blockly/workspace-content-highlight@4.0.16
 - @blockly/continuous-toolbox@5.0.18
 - @blockly/plugin-cross-tab-copy-paste@5.0.10
 - @blockly/create-package@2.0.5
 - @blockly/dev-tools@7.1.10
 - @blockly/disable-top-blocks@0.4.14
 - @blockly/field-angle@4.0.3
 - @blockly/field-bitmap@4.0.18
 - @blockly/field-colour@4.0.3
 - @blockly/field-colour-hsv-sliders@4.2.6
 - @blockly/field-date@8.0.16
 - @blockly/field-dependent-dropdown@3.0.16
 - @blockly/field-grid-dropdown@4.0.15
 - @blockly/field-multilineinput@4.0.3
 - @blockly/field-slider@6.1.9
 - @blockly/fixed-edges@4.0.15
 - @blockly/keyboard-navigation@0.5.12
 - @blockly/migrate@2.0.6
 - @blockly/plugin-modal@6.0.14
 - @blockly/plugin-scroll-options@5.0.17
 - @blockly/shadow-block-converter@5.0.5
 - @blockly/plugin-strict-connection-checker@4.0.15
 - @blockly/suggested-blocks@4.1.10
 - @blockly/theme-dark@6.0.7
 - @blockly/theme-deuteranopia@5.0.8
 - @blockly/theme-highcontrast@5.0.7
 - @blockly/theme-modern@5.0.7
 - @blockly/theme-tritanopia@5.0.7
 - @blockly/toolbox-search@1.2.10
 - @blockly/plugin-typed-variable-modal@7.0.14
 - @blockly/workspace-backpack@5.3.8
 - @blockly/workspace-minimap@0.1.18
 - @blockly/plugin-workspace-search@8.1.7
 - @blockly/zoom-to-fit@5.0.17

maribethb committed:
fix: remove keyboard nav option from dev-tools (#2347)

cpcallen committed:
fix(block-test): Fix incorrect references to Blockly.prompt (#2344)

This is now Blockly.dialog.prompt, and has been since
7.20211209.0.

cpcallen committed:
chore: Merge branch 'master' into v11.0.0

maribethb committed:
fix: gh pages formatting (#2343)

maribethb commented on plugins/dev-tools/src/screenshot.js:
```diff
@@ -58,7 +58,7 @@ function workspaceToSvg_(workspace, callback, customCss) {
   const textAreas = document.getElementsByTagName('textarea');
   for (let i = 0; i < textAreas.length; i++) {
-    textAreas[i].innerHTML = textAreas[i].value;
+    textAreas[i].innerText = textAreas[i].value;
```
Not sure where exactly the conflict was, but I think this is supposed to be `textContent` instead of `innerText` based on https://github.com/google/blockly-samples/pull/2328#issuecomment-2066987077

not sure if it did get updated and that was lost in the merge conflict, or if it never did.

maribethb reviewed (approved):
LGTM, CLA failures are our own bots and expected/allowed.

cpcallen committed:
chore: Run npm install to update package lock files

Following the previous merge commit, the state of some of the
package-lock.json files was not quite in sync with their
corresponding package.json files.  Running npm install fixed
this and resulted in the two changes in this commit.

maribethb committed:
fix: remove angle fields from block-test (#2346)

* fix: remove angle fields from block-test

* fix: remove colour label

cpcallen committed:
chore: Merge branch 'master' into rc/v11.0.0
